cdk/lib/elastic-search-monitor.ts

```diff
@@ -70,5 +76,106 @@ export class ElasticSearchMonitor extends GuStack {
+        dimensionsMap: {
+          Cluster: clusterName,
+        },
+      });
+    };
+
+    const commonAlarmProps = {
+      app,
+      snsTopicName: topicForElasticsearchAlerts.topicName,
+      period: Duration.minutes(1),
+    };
+
+    const lessThanAlarmProps = {
+      ...commonAlarmProps,
+      comparisonOperator: ComparisonOperator.LESS_THAN_THRESHOLD,
+      statistic: "Minimum",
+    };
+
+    const greaterThanAlarmProps = {
+      ...commonAlarmProps,
+      comparisonOperator: ComparisonOperator.GREATER_THAN_THRESHOLD,
+      statistic: "Maximum",
+    };
+
+    new GuAlarm(this, "DataNodeCountAlarm", {
+      ...lessThanAlarmProps,
+      alarmDescription: `Unexpected count of data nodes in ${this.stage}`,
+      evaluationPeriods: 2,
+      metric: metric("NumberOfDataNodes"),
+      threshold: 3,
+    });
+
+    new GuAlarm(this, "MasterNodeCountAlarm", {
+      ...lessThanAlarmProps,
+      alarmDescription: `Unexpected count of master nodes in ${this.stage}`,
+      evaluationPeriods: 2,
+      metric: metric("NumberOfRespondingMastersNodes"),
+      threshold: 3,
+    });
+
+    new GuAlarm(this, "ClusterStatusAlarm", {
+      ...greaterThanAlarmProps,
+      alarmDescription: `Unexpected cluster status in ${this.stage}`,
+      evaluationPeriods: 15,
+      metric: metric("Status"),
+      // Green = 0, Yellow = 1, Red = 2
+      threshold: 0,
+    });
+
+    new GuAlarm(this, "RedClusterStatusAlarm", {
+      ...greaterThanAlarmProps,
+      alarmDescription: `Red cluster status in ${this.stage}`,
+      evaluationPeriods: 1,
+      metric: metric("Status"),
+      // Green = 0, Yellow = 1, Red = 2
+      threshold: 1,
+    });
+
+    new GuAlarm(this, "DataNodeJvmHeapUsageAlarm", {
+      ...greaterThanAlarmProps,
+      alarmDescription: `A data node is using too much of its heap in ${this.stage}`,
+      evaluationPeriods: 2,
+      metric: metric("MaxJvmHeapUsage"),
+      threshold: 85,
+    });
+
+    const lowStorageDescription = `A data node is running on less than 85% disk space
+    (this is considered a low disk watermark: https://www.datadoghq.com/blog/elasticsearch-unassigned-shards/#reason-5-low-disk-watermark).
+    In order to troubleshoot this, follow the instructions on this Trello card: https://trello.com/c/68Bi7pwB.
+    Or, if you can't access that you can do the following: Find the id of the instance that is experiencing problems by
+    checking https://logs.gutools.co.uk/cerebro. SSH onto the instance ($ ssm ssh -i "instance-id" -p deployTools).
+    Double check the disk space used (curl -s 'localhost:9200/_cat/allocation?v') and access the logs to see if you can
+    find any useful error messages ($ grep "ERROR" /var/log/syslog).`;
+
+    new GuAlarm(this, "DataNodeLowStorageAlarm", {
+      ...lessThanAlarmProps,
+      alarmDescription: lowStorageDescription,
+      evaluationPeriods: 2,
+      metric: metric("MinFreeDiskSpace"),
+      threshold: 280375465082.88,
```

Comment:
Could we comment this somehow or make it less magical?

Reply:
Thanks for the suggestion. This prompted me to check what this value actually was and it's drifted out of date (presumably we changed the instance type for data nodes since the alarm was manually updated), so I recalculated and pulled it into a variable: https://github.com/guardian/elastic-search-monitor/pull/19/commits/a287236d6f2c58cac3effc599d4928970af6401d.

Perhaps it'd be easier if the application code pushed this as a percentage-based metric, but since we've reserved these instances for another year I'll leave it as it is for now!